vllm/v1/core/kv_cache_manager.py

```diff
@@ -64,7 +65,131 @@ def new_empty(self) -> "KVCacheBlocks":
         return KVCacheBlocks(tuple([] for _ in range(len(self.blocks))))
 
 
-class KVCacheManager:
+class KVCacheManagerInterface(ABC):
+
+    @property
+    @abstractmethod
+    def usage(self) -> float:
+        raise NotImplementedError
+
+    @abstractmethod
+    def make_prefix_cache_stats(self) -> Optional[PrefixCacheStats]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_computed_blocks(self,
+                            request: Request) -> tuple[KVCacheBlocks, int]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def allocate_slots(
+        self,
+        request: Request,
+        num_new_tokens: int,
+        num_new_computed_tokens: int = 0,
+        new_computed_blocks: Optional[KVCacheBlocks] = None,
+        num_draft_tokens: int = 0,
+        num_lookahead_tokens: int = 0,
+        delay_cache_blocks: bool = False,
+    ) -> Optional[KVCacheBlocks]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def free(self, request: Request) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def reset_prefix_cache(self) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_num_common_prefix_blocks(
+        self,
+        request: Request,
+        num_running_requests: int,
+    ) -> list[int]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def free_block_hashes(self, request: Request) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def take_events(self) -> list[KVCacheEvent]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_block_ids(self, request_id: str) -> tuple[list[int], ...]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def cache_blocks(self, request: Request, num_computed_tokens: int) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def create_empty_block_list(self) -> KVCacheBlocks:
+        raise NotImplementedError
+
+
+class DummyKVCacheManager(KVCacheManagerInterface):
+
+    @property
+    def usage(self) -> float:
+        return 0.0
+
+    def make_prefix_cache_stats(self) -> Optional[PrefixCacheStats]:
+        return None
+
+    def get_computed_blocks(self,
+                            request: Request) -> tuple[KVCacheBlocks, int]:
+        return (KVCacheBlocks([]), 0)
```

Comment:
What's your expect behavior of prefix caching? Seems that you want prefix caching to be enabled in "last" pooling method.
https://github.com/vllm-project/vllm/blob/8aeaa910a2c514022ab7521ce74eec0734886a1b/vllm/config.py#L4594